### PR TITLE
fix: use raw idents in `make::{name, name_ref}` when used with a keyword

### PR DESCRIPTION
**Pull request by jonas-schievink**
fixes https://github.com/rust-analyzer/rust-analyzer/issues/8680

bors r+

### CODE REVIEW

**Comment by jonas-schievink** [comment]
oh, this is going to be more complicated

File: crates/syntax/src/ast/make.rs [hunk 1]
```diff
@@ -15,12 +15,22 @@ use stdx::format_to;
...
 }
 
 pub fn name_ref(text: &str) -> ast::NameRef {
-    ast_from_text(&format!("fn f() {{ {}; }}", text))
+    ast_from_text(&format!("fn f() {{ {}{}; }}", raw_ident_esc(text), text))
 }
+
+fn raw_ident_esc(ident: &str) -> &'static str {
+    let is_keyword = parser::SyntaxKind::from_keyword(ident).is_some();
+    if is_keyword && !matches!(ident, "self" | "crate" | "super" | "Self") {
```
Comment (matklad) [inline crates/syntax/src/ast/make.rs:27]: Seems that `Self` is an odd one out here? (or `from_keyword` is really weird?)

Comment (jonas-schievink) [reply crates/syntax/src/ast/make.rs:27]: Yeah, these are taken from https://doc.rust-lang.org/reference/identifiers.html#identifiers

**Pull request merged**